tests/test_testbench_retry.py

```diff
@@ -655,6 +655,107 @@ def test_retry_test_return_error_after_bytes(self):
         self.assertIn("size", create_rest)
         self.assertEqual(int(create_rest.get("size")), 2 * UPLOAD_QUANTUM)
 
+    def test_write_retry_test_stall_after_bytes(self):
+        # Create a new bucket
+        response = self.client.post(
+            "/storage/v1/b", data=json.dumps({"name": "bucket-name"})
+        )
+        self.assertEqual(response.status_code, 200)
+
+        # Setup a stall for reading back the object.
+        response = self.client.post(
+            "/retry_test",
+            data=json.dumps({
+                "instructions": {
+                    "storage.objects.insert": [
+                        "stall-for-1s-after-250K",
+                        "stall-for-1s-after-300K"
+                    ]
+                }
+            }),
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(response.headers.get("content-type").startswith("application/json"))
+
+        create_rest = json.loads(response.data)
+        self.assertIn("id", create_rest)
+        test_id = create_rest.get("id")
+
+        # Initiate resumable upload
+        response = self.client.post(
+            "/upload/storage/v1/b/bucket-name/o",
+            query_string={"uploadType": "resumable", "name": "stall"},
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+
+        location = response.headers.get("location")
+        self.assertIn("upload_id=", location)
+        match = re.search(r"[&?]upload_id=([^&]+)", location)
+        self.assertIsNotNone(match, msg=location)
+        upload_id = match.group(1)
+
+        # Upload the first 256KiB chunk of data and trigger the stall.
```

Comment:
The only test I don't see is stall with incremental progress from a multiple of quantum

1. Start upload
2. Upload `UPLOAD_QUANTUM * 2` stall after first quantum; assuming you would end connection in retries and query service for offset
3. Retry upload from offset `UPLOAD_QUANTUM-1`

Reply:
This is scoped for write stall support under `stall-xs-after-yk` but outside of scope for this PR; thanks for clarifying @Tulsishah

Reply:
As discussed offline this is not in the scope of this PR